eng/common/docgeneration/Generate-DocIndex.ps1

```diff
@@ -126,7 +126,7 @@ function GenerateDocfxTocContent([Hashtable]$tocContent, [String]$lang, [String]
     New-Item -Path $YmlPath -Name "toc.yml" -Force
     $visitedService = @{}
     # Sort and display toc service name by alphabetical order, and then sort artifact by order.
-    $sortedToc = $tocContent.Values | Sort-Object NewIndex, TypeIndex, ServiceName, DisplayName, Artifact
+    $sortedToc = $tocContent.Values | Sort-Object ServiceName, NewIndex, TypeIndex, DisplayName, Artifact
```

Comment:
Is this only used by the ToC? Where does the sorting for the package list on a given page happen?

Reply:
Our workflow is like,
loop in package in the defined order:
If we first meet the service,
  1. Create {service}.md and put the package info into {service}.md
  2. Then put the service entry into toc.yml.
If the service visited before,
 1. Add content to {service}.md
So we have to sort the service to have service sorted in toc.yml first, then in each {service}.md we sort by track 2, client type and display name.